src/test/java/org/folio/rest/api/CheckInStorageApiTest.java

```diff
@@ -0,0 +1,194 @@
+package org.folio.rest.api;
+
+import static org.folio.rest.support.matchers.ValidationErrorMatchers.hasMessage;
+import static org.folio.rest.support.matchers.ValidationErrorMatchers.hasParameter;
+import static org.folio.rest.support.matchers.ValidationResponseMatchers.isValidationResponseWhich;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.text.MatchesPattern.matchesPattern;
+import static org.junit.Assert.assertTrue;
+
+import java.net.MalformedURLException;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.folio.rest.support.ApiTests;
+import org.folio.rest.support.IndividualResource;
+import org.folio.rest.support.JsonResponse;
+import org.folio.rest.support.MultipleRecords;
+import org.folio.rest.support.builders.CheckInBuilder;
+import org.folio.rest.support.http.AssertingRecordClient;
+import org.folio.rest.support.http.InterfaceUrls;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.vertx.core.json.JsonObject;
+
+public class CheckInStorageApiTest extends ApiTests {
+  private static final String TABLE_NAME = "check_in";
+
+  private final AssertingRecordClient checkInClient =
+    new AssertingRecordClient(client, StorageTestSuite.TENANT_ID,
+      InterfaceUrls::checkInsStorageUrl, "checkIns");
+
+  @Before
+  public void beforeEach() {
+    StorageTestSuite.cleanUpTable(TABLE_NAME);
+  }
+
+  @After
+  public void checkIdsAfterEach() {
+    StorageTestSuite.checkForMismatchedIDs(TABLE_NAME);
+  }
+
+  @Test
+  public void canCreateCheckIn() throws InterruptedException,
+    ExecutionException, TimeoutException, MalformedURLException {
+
+    JsonObject checkInToCreate = createSampleCheckIn().create();
+
+    IndividualResource createResult = checkInClient.create(checkInToCreate);
+
+    assertThat(createResult.getJson(), is(checkInToCreate));
+  }
+
+  @Test
+  public void canCreateRecordWithoutId() throws InterruptedException,
+    ExecutionException, TimeoutException, MalformedURLException {
+
+    JsonObject checkInToCreate = createSampleCheckIn()
+      .withId(null)
+      .create();
+
+    IndividualResource createResult = checkInClient.create(checkInToCreate);
+
+    assertThat(createResult.getId(), notNullValue());
+
+    checkInToCreate.put("id", createResult.getId());
```

Comment:
Why is an `id` added to the record after creation?

Reply:
In order to simplify comparison - compare json only instead of field by field.

Reply:
I think that comes with a trade off between this reuse and the clarity of the tests. I think this obscures the intent that the `id` is server generated.

Maybe an alternative would be to to check the record (inc not null ID) except the ID value within the shared method and check the ID separately in the relevant tests.

Reply:
I've added matcher that will compare only those properties that present on both records.